src/lib/rng/auto_rng/auto_rng.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+* (C) 2016 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/auto_rng.h>
+#include <botan/entropy_src.h>
+#include <botan/loadstor.h>
+#include <botan/internal/os_utils.h>
+
+#if defined(BOTAN_HAS_HMAC_DRBG)
+  #include <botan/hmac_drbg.h>
+#endif
+
+#if defined(BOTAN_HAS_HMAC_RNG)
+  #include <botan/hmac_rng.h>
+#endif
+
+#if defined(BOTAN_HAS_SYSTEM_RNG)
+  #include <botan/system_rng.h>
+#endif
+
+namespace Botan {
+
+AutoSeeded_RNG::AutoSeeded_RNG(size_t max_output_before_reseed)
+   {
+   m_rng.reset(new BOTAN_AUTO_RNG_DRBG(MessageAuthenticationCode::create(BOTAN_AUTO_RNG_HMAC),
+#if defined(BOTAN_HAS_SYSTEM_RNG)
+                                       system_rng(),
+#else
+                                       Entropy_Sources::global_sources(),
+#endif
```

Comment:
@neusdan Would this work for you (just requires the constructor modification I mentioned):

```
Entropy_Sources my_entropy({"rdrand","rdseed","win32_stats"});
// seeds from both system rng and specified entropy sources
HMAC_DRBG my_drbg("SHA-256", system_rng(), my_entropy, my_reseed_interval);
my_drbg.randomize_with_input(...);
```

The `hres_timer` was removed, the idea is such timestamps are instead provided using the `additional_data` field to the DRBG. But currently this is done only by `AutoSeeded_RNG::randomize`, so unless you copied that code to format the timestamps to a buffer and called `randomize_with_input` on `my_drbg`, no additional data is provided to the DRBG.